optuna/multi_objective/samplers/_nsga2.py

```diff
@@ -259,11 +260,12 @@ def _select_elite_population(
     def _select_parent(
         self,
         study: "multi_objective.study.MultiObjectiveStudy",
-        population: List["multi_objective.trial.FrozenMultiObjectiveTrial"],
+        population: Sequence["multi_objective.trial.FrozenMultiObjectiveTrial"],
```

Comment:
Using `Sequence` since the list is never actually modified (Liskov's principle).